src/Atmos/Model/AtmosModel.jl

```diff
@@ -499,18 +502,20 @@ equations.
     t::Real,
     direction,
 )
+
     flux_pad = SVector(1, 1, 1)
-    ts = recover_thermo_state(m, state, aux)
     tend = Flux{FirstOrder}()
-    args = (m, state, aux, t, ts, direction)
-    flux.ρ = Σfluxes(eq_tends(Mass(), m, tend), args...) .* flux_pad
-    flux.ρu = Σfluxes(eq_tends(Momentum(), m, tend), args...) .* flux_pad
-    flux.ρe = Σfluxes(eq_tends(Energy(), m, tend), args...) .* flux_pad
-
-    flux_first_order!(m.moisture, m, flux, state, aux, t, ts, direction)
-    flux_first_order!(m.precipitation, m, flux, state, aux, t, ts, direction)
-    flux_first_order!(m.tracers, m, flux, state, aux, t, ts, direction)
-    flux_first_order!(m.turbconv, m, flux, state, aux, t, ts, direction)
+    _args = (state = state, aux = aux, t = t, direction = direction)
+    args = merge(_args, (precomputed = precompute(m, _args, tend),))
```

Comment:
Any reason for `precomputed` to be specifically labelled? We could simply merge whatever `precompute` returns into `args` directly?

Reply:
Also, since you're touching this code, let's please use `atmos::AtmosModel` consistently -- it is `m::AtmosModel` in some places and that's very annoying. 😱

Reply:
> Any reason for precomputed to be specifically labelled? We could simply merge whatever precompute returns into args directly?

I kind of like the idea of encapsulating the pre-computed things separately from the top-level arguments (which is basically what `_args` is), so that we know what is cached vs what is a top-level argument/positional arg when we're deep in the tendency kernels.

> Also, since you're touching this code, let's please use atmos::AtmosModel consistently -- it is m::AtmosModel in some places and that's very annoying. 😱

Yes, I can update that. It's been bothering me also.